### PR TITLE
Add ignoreUndefinedProperties

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Port of https://github.com/googleapis/nodejs-firestore/pull/1062

### CODE REVIEW

File: packages/firebase/index.d.ts [hunk 1]
```diff
@@ -7743,6 +7743,15 @@ declare namespace firebase.firestore {
      * @webonly
      */
     experimentalForceLongPolling?: boolean;
+
+    /**
+     * Whether to skip nested properties that are set to `undefined` during
+     * object serialization. If set to `true`, these properties will be skipped
+     * and are not be written to Firestore. If the setting is set `false` or
```
Comment (wu-hui) [inline packages/firebase/index.d.ts:7750]: s/the setting is set/it is set to

Comment (schmidt-sebastian) [reply packages/firebase/index.d.ts]: Done

File: packages/firebase/index.d.ts [hunk 2]
```diff
@@ -7743,6 +7743,15 @@ declare namespace firebase.firestore {
      * @webonly
      */
     experimentalForceLongPolling?: boolean;
+
+    /**
+     * Whether to skip nested properties that are set to `undefined` during
+     * object serialization. If set to `true`, these properties will be skipped
```
Comment (egilmorez) [inline packages/firebase/index.d.ts:7749]: Very much a nit, but we prefer present over future when possible.  Here it cold be "properties are skipped . . . . the SDK throws . . ."

Comment (schmidt-sebastian) [reply packages/firebase/index.d.ts]: Updated. Thanks for the review!

**Pull request merged**